### PR TITLE
[Test] Add tests for learning moves via form change

### PR DESCRIPTION
**Pull request by PigeonBar**
## Why am I making these changes?

There were no tests for this.

## What are the changes from a developer perspective?

Title.

## How to test the changes?

`npm run test form-change-phase`

## Checklist

- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes manually?
- [x] Are all unit tests still passing? (`npm run test:silent`)
  - [x] Have I created new automated tests (`npm run create-test`) or updated existing tests related to the PR's changes?

### CODE REVIEW

File: test/phases/form-change-phase.test.ts [hunk 1]
```diff
@@ -70,5 +73,49 @@ describe("Form Change Phase", () => {
...
+    // Manually trigger a form change
+    game.scene.unshiftPhase(new FormChangePhase(rillaboom, pokemonFormChanges[Species.RILLABOOM][1], false));
+
+    game.move.use(MoveId.SPLASH);
+    await game.toNextTurn();
+
+    // After the form change: Should be normal form
+    expect(game.phaseInterceptor.log.includes("FormChangePhase")).toBe(true);
+    expect(rillaboom.getFormKey()).toBe("");
+    expect(rillaboom.moveset.map((m) => m.moveId)).not.toContain(MoveId.G_MAX_DRUM_SOLO);
```
Comment (Tempo-anon) [inline test/phases/form-change-phase.test.ts:119]: Would it still try to learn drum beating here?

**Review by torranx — APPROVED** [review]

File: test/phases/form-change-phase.test.ts (hunk 1 above)
Comment (PigeonBar) [reply test/phases/form-change-phase.test.ts:119]: It does try to learn Drum Beating. I guess this is incorrect behaviour?

Comment (Tempo-anon) [reply test/phases/form-change-phase.test.ts:119]: Hm... I think that's just a weird edge case. Maybe people will constantly deactivate and reactivate to change movesets? I don't know how important that is but the code in form-change-phase can probably be modified to only do the move learning if the user's new form index isn't 0 since all form change learn moves are g-maxes or zacian/zamazenta

**Review by Tempo-anon — APPROVED** [review]
